### PR TITLE
Add docs to the CRM App

### PR DESCRIPTION
**Pull request by lkostrowski**

### CODE REVIEW

File: docs/developer/app-store/apps/crm.mdx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+- Initial index of existing customers stored in Saleor and sending them to the provider contact list
+- Creating a new contact in the provider, when a new Customer is created or updated in Saleor
+
+:::caution
+
+To successfully configure the CRM App, you must:
+
+- Have a Mailchimp account
+- Have `MANAGE_APPS` permission
```
Comment (peelar) [inline docs/developer/app-store/apps/crm.mdx:26]:
```suggestion
- Have a Saleor user with `MANAGE_APPS` permission
```

File: docs/developer/app-store/apps/crm.mdx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+:::
+
+## Installation
+
+The App can be installed from Saleor App Store. To install the App, you must have `MANAGE_USERS` and `MANAGE_APPS` permissions.
```
Comment (peelar) [inline docs/developer/app-store/apps/crm.mdx:32]:
```suggestion
The App can be installed from [Saleor App Store](developer/app-store/overview.mdx). To install the App, you must have `MANAGE_USERS` and `MANAGE_APPS` [permissions](developer/permissions.mdx#app-permissions).
```

Also, can you add the same section to the Taxes app? 🙏

File: docs/developer/app-store/apps/crm.mdx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+## Configuration
+
+The App is configured per provider. You can configure more than one provider - the App will work with at least one provider set up
```
Comment (peelar) [inline docs/developer/app-store/apps/crm.mdx:36]:
```suggestion
The App is configured per provider. You can configure multiple providers - the App will work with at least one provider set up.
```

> The App is configured per provider.  

This sentence is a bit unclear to me. When I first read it, I thought it meant that each app could have only one provider, but it is denied in the next sentence.

File: docs/developer/app-store/apps/crm.mdx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+### Mailchimp configuration
```
Comment (peelar) [inline docs/developer/app-store/apps/crm.mdx:38]:
```suggestion
### Mailchimp
```

File: docs/developer/app-store/apps/crm.mdx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+#### Authorization
+
+App uses Mailchimp OAuth flow, meaning it will ask you to log in with Mailchimp and connect the App with your Saleor account. You can always disconnect the App from Mailchimp dashboard.
+
+After successful authorization, App will store a token used to connect with a Mailchimp API.
```
Comment (peelar) [inline docs/developer/app-store/apps/crm.mdx:44]:
```suggestion
After successful authorization, the App will store a token used to connect with a Mailchimp API.
```

File: docs/developer/app-store/apps/crm.mdx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+You can disconnect your Mailchimp account and "forget" the token from App/Saleor from the App configuration screen. The token will be removed and you will have to connect with Mailchimp again to continue using it.
+
+To permanently disconnect the App, you have to visit the Mailchimp dashboard.
+
+#### Customer creation webhook
+
+The App allows sending a new or updated customers to Mailchimp via a dedicated Saleor webhook. By default, this feature is disabled.
+When you enable it, you have to select what Mailchimp [audience list](https://mailchimp.com/developer/marketing/api/lists/) contact should be synchronized to (likely you have only one).
+
+The App will set a customer in Mailchimp when customer is created (`CUSTOMER_CREATED` event) and updated (`CCUSTOMER_UPDATED` event).
```
Comment (peelar) [inline docs/developer/app-store/apps/crm.mdx:55]:
```suggestion
The App will set a customer in Mailchimp when a Saleor customer is created (`CUSTOMER_CREATED` event) or updated (`CUSTOMER_UPDATED` event).
```

File: docs/developer/app-store/apps/crm.mdx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+#### Customers bulk sync
+
+If you install the App and already have previously created customers, you would likely want to perform initial index of customers and send the to the provider.
+
+You can do that from the App.
```
Comment (peelar) [inline docs/developer/app-store/apps/crm.mdx:61]: This sentence feels like something is missing here. I suggest providing some more context about where and how the user can do it.

File: docs/developer/app-store/apps/crm.mdx [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+:::caution
+
+Customers synchronization **works on the frontend side**, that means App **must remain opened** during this operation.
+In large shops this operation can take a few seconds.
+
+:::
+
+#### Mailchimp tags and fields mapping
+
+App will create contacts with "Non-subscribed" status (`transactional` in the API).
```
Comment (peelar) [inline docs/developer/app-store/apps/crm.mdx:72]:
```suggestion
The contacts created by the App will have the "Non-subscribed" status (`transactional` in the API).
```

File: docs/developer/app-store/apps/crm.mdx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+App will automatically map the customer email (mandatory), first (`FNAME`) and last (`LNAME`) names (optional)
+
+#### Tags
+
+Your campaigns likely rely on segments based on custom fields and tags.
+We probably need consent to run marketing for each user.
+
+The App allows this workflow via tags in the customer's [PrivateMetadata](https://docs.saleor.io/docs/3.x/developer/metadata)
+
+The App will parse metadata items with the key `mailchimp_tags` and extract tags that will be added to each contact.
```
Comment (peelar) [inline docs/developer/app-store/apps/crm.mdx:85]:
```suggestion
The App will parse metadata items with the `mailchimp_tags` key and extract tags that will be added to each contact.
```

File: docs/developer/app-store/apps/crm.mdx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+
+1. Follow the [_Setup_ section in the _Development_ article](developer/app-store/development.mdx#setup).
+2. Go to the app directory.
+3. Copy the `.env.example` file to `.env`.
+
+`MAILCHIMP_CLIENT_ID` and `MAILCHIMP_CLIENT_SECRET` can be obtained in Mailchimp Dashboard when new OAuth integration is created.
+
+:::info
+
+Taxes is a Next.js application. If you want to learn more about setting environment variables in Next.js, head over to the [documentation](https://nextjs.org/docs/basic-features/environment-variables).
```
Comment (peelar) [inline docs/developer/app-store/apps/crm.mdx:120]:
```suggestion
CRM is a Next.js application. If you want to learn more about setting environment variables in Next.js, head over to the [documentation](https://nextjs.org/docs/basic-features/environment-variables).
```

File: docs/developer/app-store/apps/index.mdx [hunk 11]
```diff
@@ -28,6 +28,8 @@ Saleor App Store contains the following apps:
 
 [**Taxes** Calculate order and checkout taxes using external services.](taxes)
 
+[**CRM** Synchronize customers with CRM platforms](taxes)
```
Comment (peelar) [inline docs/developer/app-store/apps/index.mdx:31]:
```suggestion
[**CRM** Synchronize customers with CRM platforms](crm)
```

**Review by peelar — CHANGES_REQUESTED** [review]

**Comment by 2can** [comment]
@peelar @lkostrowski deployment issues has been fixed with https://github.com/saleor/saleor-docs/pull/705, please rebase

**Review by peelar — APPROVED** [review]

**Review by timuric — APPROVED** [review]

**Pull request merged**